projects/observability/src/pages/explorer/explorer-service.ts

```diff
@@ -17,7 +18,9 @@ export class ExplorerService {
   ) {}
   public buildNavParamsWithFilters(
     scopeQueryParam: ScopeQueryParam,
-    filters: ExplorerDrilldownFilter[]
+    filters: ExplorerDrilldownFilter[],
+    startTime?: Date,
+    endTime?: Date
```

Comment:
why not use `TimeRange` here? It will also solve twice the `null` check at L53.
Any particular reason for making it generic?

Reply:
oh yeah, updated